src/simulator/ackermann_model.cpp

```diff
@@ -93,6 +103,38 @@ void AckermannModel::Step(const double &dt) {
   // Update the Pose
   pose_.translation += Eigen::Rotation2Df(pose_.angle) * d_vector;
   pose_.angle = AngleMod(pose_.angle + dtheta);
+  this->Transform();
+}
+
+void AckermannModel::Transform() {
+  Eigen::Rotation2Df R(math_util::AngleMod(pose_.angle));
+  Eigen::Vector2f T = pose_.translation;
+
+  for (size_t i=0; i < template_lines_.size(); i++) {
+    pose_lines_[i].p0 = R * (template_lines_[i].p0) + T;
+    pose_lines_[i].p1 = R * (template_lines_[i].p1) + T;
+  }
 }
 
+void AckermannModel::SetTemplateLines(const float r, const int num_segments){
+  // copied directly from human model. In future refractor the code to 
```

Comment:
If this is something that is needed by every robot_model and the human_model, then the correct thing to do is add the necessary code to entity_base, which they all already inherit from.

Reply:
Currently, there is a lot of similarity between robots and humans as they both are represented by a cylinder. After thinking through this, I think it is best to separate the code as if we decide, for example, to use a more accurate shape representing humans that change would not affect robot_model.